test/cbmc/proofs/OTA_MQTT_strerror/OTA_MQTT_strerror_harness.c

```diff
@@ -0,0 +1,40 @@
+/*
+ * AWS IoT Over-the-air Update v3.0.0
+ * Copyright (C) 2020 Amazon.com, Inc. or its affiliates.  All Rights Reserved.
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy of
+ * this software and associated documentation files (the "Software"), to deal in
+ * the Software without restriction, including without limitation the rights to
+ * use, copy, modify, merge, publish, distribute, sublicense, and/or sell copies of
+ * the Software, and to permit persons to whom the Software is furnished to do so,
+ * subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in all
+ * copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY, FITNESS
+ * FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR
+ * COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER
+ * IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE SOFTWARE.
+ */
+
+/**
+ * @file OTA_MQTT_strerror_harness.c
+ * @brief Implements the proof harness for OTA_MQTT_strerror function.
+ */
+
+#include "ota_mqtt_private.h"
+
+void OTA_MQTT_strerror_harness()
+{
+    OtaMqttStatus_t status;
+    const char * pErrorMsg;
+
+    pErrorMsg = OTA_MQTT_strerror( status );
+
+    /* The function returns a string which is never NULL. If it does, then it is a
+     *  problem. */
```

Comment:
It's true that based on the implementation, the string can not be returned as a non-NULL value. Kind of a nitpick, but the assertion is more about the expectation of the caller and less about the implementation. The assert is to confirm that the implementation is doing what we expect it to be doing. So I would reword it to be something like this:
```suggestion
    /* The caller of this function assumes that the return value is never NULL. */
```

Reply:
Updated it in 207cf11